proposals/model-management.md

```diff
@@ -0,0 +1,67 @@
+# Model Management
+
+## Problem
+
+This proposal is attempting to solve the following problems:
+
+* Slow model loading on cold-start
+* Storage of LoRA adapters generated from finetuning jobs
+* Dynamic LoRA adapter injection
+
+While working within the following constraints:
+
+* KubeAI should be able to run with zero external dependencies
+* KubeAI should be able to run on a local machine
+* KubeAI should be able to be run on clouds with heterogenous disk features (limited support for ReadOnlyMany)
+
+## Solution
+
+KubeAI should support configurable storage for both models and adapters. These settings should be configurable at the `kind: Model` level as some variants (like GCP Hyperdisk ML) are expensive and generally only recommended for larger models.
+
+```yaml
+kind: Model
+spec:
+  storageProfile:
+    model: Bucket # || None (Always Pull) || ReadWriteManyDisk (EFS/NFS) || ReadManyDisk (GCP ML Hyperdisk)
+    adapters: Bucket # || ReadWriteManyDisk (EFS/NFS)
+```
+
+Storage profiles should be defined at the KubeAI config.yaml level:
+
+```yaml
+storageProfiles:
+  Bucket:
+    bucket:
+      url: s3://...
+      # ...
+  ReadWriteManyDisk:
+    disk:
+      storageClass: aws-efs
+      # ...
+  ReadManyDisk:
+    disk:
+      storageClass: gcp-hyperdisk
+      # ...
+  # While standard storage profile names are recommended, they should be open ended
+  # to allow for multiple different admin-defined varieties:
+  MyCustomBucket:
+    bucket:
+    # ...
+```
+
+### Bucket storage
+
+This mode relies on access to a bucket. To support zero external dependencies and local deployment this bucket could be provided by a Helm-deployed Minio (possible License issue?) / [SeaweedFS](https://github.com/seaweedfs/seaweedfs) (mature/secure?) instance.
```

Comment:
Seaweed would be my choice. I looked into Seaweedfs in the past for same use case. Seaweed is lightweight and has a single binary option.